script.js

```diff
@@ -1,62 +1,106 @@
-/* eslint-disable no-unused-vars */
-const booksContainer = document.querySelector('#books-container');
-const booksForm = document.querySelector('.books-form');
-const bookTitle = document.getElementById('title');
-const bookAuthor = document.getElementById('author');
+// Refactoring using classes
 
-const booksArr = [];
+/* eslint-disable max-classes-per-file */
 
-// local storage
-function booksStorage() {
-  localStorage.setItem('books', JSON.stringify(booksArr));
+class CreateBook {
+  constructor(title, author, id) {
+    this.title = title;
+    this.author = author;
+    this.id = id;
+  }
 }
-// add book to list
-function addBook(title, author) {
-  const newBook = {
-    title,
-    author,
-  };
-  booksArr.push(newBook);
-
-  const bookItem = document.createElement('div');
-  booksArr.forEach((book, index) => {
-    bookItem.innerHTML = `
-    <div class="book-item">
-    <p class="book-title">${book.title}</p>
-    <p class="book-author">${book.author}</p>                                          
-    <button type="button" id="${index}" class="remove-btn">Remove</button>   
-    </div>
-    <hr>`;
-
-    const removeBtn = bookItem.querySelector('.remove-btn');
-    removeBtn.addEventListener('click', () => {
-      booksArr.splice(index, 1);
-      bookItem.remove();
-      booksStorage();
-    });
-  });
-  booksStorage();
-
-  booksContainer.appendChild(bookItem);
+
+// localStorage classes
+
+class LocalStorageClass {
+  static getbooksFromStore() {
+    let books;
+    if (localStorage.getItem('books') === null) {
+      books = [];
+    } else {
+      books = JSON.parse(localStorage.getItem('books'));
+    }
+    return books;
+  }
+
+  static addbookToStore(book) {
+    const books = LocalStorageClass.getbooksFromStore();
+    books.push(book);
+    localStorage.setItem('books', JSON.stringify(books));
+  }
+
+  static removeFromTheStore(title) {
+    const books = LocalStorageClass.getbooksFromStore();
+    const filteredArray = books.filter((book) => book.id === title);
+    localStorage.setItem('books', JSON.stringify(filteredArray));
+  }
```

Comment:
- [ ] When a single book is removed, all the store vanishes. You should be making the test between `book.id` and `id `instead of `title`
```js
 const filteredArray = books.filter((book) => book.id === id);
```